### PR TITLE
Release of SLIM Python Starterkit Sample Project

### PR DESCRIPTION
**Pull request by jpl-jengelke**
## Purpose
- Release of a unified Python sample project starterkit that automates build, release and publishing to PyPi for OSS consumption.
## Proposed Changes
- **NOTE:** Build instructions are inline with the [project README](https://github.com/NASA-AMMOS/slim-starterkit-python/blob/develop/README.md).
- Introduction of a feature rich Python project template that provides the following:
  - Modern build tooling using `pyproject.toml`, `setup.cfg` and compliant with PEP517/518 standards
  - Builds controlled outside code through parameterized configurations, using an IoC pattern
  - Automatic versioning from Git tags and commits, including incremental versioning in development
    - Releases are based on a version file in the top-level application module. 
    - To release, update the file with the proposed version and tag to that version in a release.
  - GitHub Action publishing end-to-end with automatic detection of tagged releases.
    - Tag and release and the project is sent to PyPi without further manual intervention.
  - Automatic package find, so no more manually entering packages in the build configurations!
  - New tooling, integration with:
    - The `twine`, Python `build` module and other tooling that support legacy `setuptools` behavior -- Best of both worlds!
    - Instructions for both local development installations using `pip -e` and build/package installations from packaged deliverables.
## Issues
- NASA-AMMOS/slim#46: This is the culmination of the Python starterkit template.
## Testing
- Local build and installation
- Successful execution of tag/release cycle per a mock [release on Test PyPi](https://test.pypi.org/project/slim-sample-project/)
- Validation of expectations in [GitHub Actions build cycle](https://github.com/NASA-AMMOS/slim-starterkit-python/actions/runs/3699395992/jobs/6266716481)

### CODE REVIEW

**Comment by jpl-jengelke** [comment]
Please review as the final PR is ready.

**Comment by jpl-jengelke** [comment]
https://github.com/NASA-AMMOS/slim-starterkit-python/pull/8

**Comment by jpl-jengelke** [comment]
The PR for Dependabot instructions is ready at https://github.com/NASA-AMMOS/slim/pull/84 
(In context: https://github.com/NASA-AMMOS/slim/blob/issue_73/continuous-testing/starter-kits/README.md )

The PR for using this in CI scenarios is pending shortly.

File: .github/workflows/python-publish.yml
Comment (riverma) [inline .github/workflows/python-publish.yml:35]: Nice! Good we’re templating this in here.

File: CODE_OF_CONDUCT.md
Comment (riverma) [inline CODE_OF_CONDUCT.md:62]: Might be good to reference (in your written guide, not here) that files like `CONTRIBUTING.md` need to be modified where ever you see `[INSERT..` language. See step 3 in https://nasa-ammos.github.io/slim/documentation/starter-kits/#contributing-guide-template

File: slim_sample_project/hello_world.py
Comment (riverma) [inline slim_sample_project/hello_world.py:1]: Nice - thanks for making this clear.

File: AUTHORS.md
Comment (riverma) [inline AUTHORS.md]: This file is less common nowadays due to the wide spread use of VCSes that commit and tag code contributions from people automatically. Thus you can easily see who wrote what as long as someone clones or uses code templates. I’m just worried having this file might discourage use of the starter kit if people see names they don’t recognize and wonder if they can use code here or need to check for permission.

What are your thoughts?

**Review by riverma — COMMENTED** [review]
Looks really good! I wanted to test run this and will wait to do that when the SLIM write-up is ready. Excellent stuff @jpl-jengelke!

File: .github/workflows/python-publish.yml
Comment (jpl-jengelke) [reply .github/workflows/python-publish.yml:35]: Thanks. This is notable because it must be mentioned in Python instructions.

File: AUTHORS.md
Comment (jpl-jengelke) [reply AUTHORS.md]: Sure, I have no issues removing it.

**Comment by jpl-jengelke** [comment]
Based on feedback in the main project, there are no additional changes necessary here. Waiting on approval to merge.

**Review by sharimayer — APPROVED** [review]

**Review by riverma — APPROVED** [review]
Go for merge 🚀. Thanks @jpl-jengelke!

**Pull request merged**
